### PR TITLE
Override document elements after document.write (close #253)

### LINKED ISSUE #253: The onDocumentCleaned event is raised wrongly after document.write is called for a document that is still being loaded
Multiple calls to  `document.write`  leads to the page hanging. Because each time`onDocumentCleaned` is raised Hammerhead gets initialized.

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @inikulin @churkin

### CODE REVIEW

File: test/client/fixtures/sandbox/node/document-test.js [hunk 1]
```diff
@@ -274,3 +274,20 @@ asyncTest('the onDocumentCleaned event is not raised after calling document.writ
 
     $('<iframe></iframe>').attr('src', iframeSrc).appendTo('body');
 });
+
+asyncTest('should document elements be overridden after document.write (GH-253)', function () {
```
Comment (inikulin) [inline test/client/fixtures/sandbox/node/document-test.js]: We don't use BDD-style names in qunit tests. Check with @VasilyStrelyaev

**Comment by inikulin** [comment]
@LavrovArtem So, this fixes yesterday regression in HHM runs with `JSON.parse`, right?

**Comment by LavrovArtem** [comment]
@inikulin yep

File: test/client/data/node-sandbox/iframe-ovverride-elems-after-write.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+</script>
+<script>
+    var iframeSandbox = window['%hammerhead%'].sandbox.iframe;
+    var INTERNAL_PROPS = hammerhead.get('../processing/dom/internal-properties');
+
+    iframeSandbox.on(iframeSandbox.IFRAME_READY_TO_INIT_EVENT, window.top.initIframeTestHandler);
+
+    document.write('<a>Link</a><iframe id="test"></iframe>');
+
+    window.top.strictEqual(document.querySelector('a')[INTERNAL_PROPS.processedContext], window);
```
Comment (inikulin) [inline test/client/data/node-sandbox/iframe-ovverride-elems-after-write.html]: It's really bad idea to move assertions out of the test body. Can we avoid it?

Comment (LavrovArtem) [inline test/client/data/node-sandbox/iframe-ovverride-elems-after-write.html]: postMessage

Comment (inikulin) [inline test/client/data/node-sandbox/iframe-ovverride-elems-after-write.html]: Let's do it then

**Comment by LavrovArtem** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by churkin** [comment]
iframe-ovverride-elems-after-write.html -> iframe-override-elems-after-write.html

**Comment by churkin** [comment]
\r

**Comment by LavrovArtem** [comment]
FPR

**Comment by churkin** [comment]
lgtm

**Pull request merged**